src/i18n/README.md

```diff
@@ -1,3 +1,49 @@
 This is localisation layer, that allows to translate the app into different languages.
 
 All messages should be stored in a messages.json, using that format https://www.i18next.com/misc/json-format#i18next-json-v4
+
+### How to use
+
+1. Using your project i18n tooling, construct an object with localised messages, like:
+```
+// You can use https://ttag.js.org/ for example:
+const localised = {
+  login: t`Log in`,
+  ...
+}
+```
+
+2. Pass it to i18n singleton, using `setTranslations`. That would provide localisation to non-react components, like Auth.
+
+```
+// Your messages object should have loalised strings at that moment
+const localised = {
+  login: 'Inloggen',
+  ...
+}
+
+setTranslations(localised);
+```
+
+3. [Optional] Add I18nContextHolder into your root tree. 
+   If you skip this step, messages from singleton service would be used.
+   Note: setTranslations() calls won't cause React tree rerender, so you have to call it before first render.
```

Comment:
I don't think it would work. If an user doesn't render `I18nContextHolder`, the default value of `messages` in the context will never be overwritten by a `setTranslations` call, because the context will be initialized when the `i18n-context.ts` module is imported and we cannot call `setTranslations` before that. I created a minimal reproducible example https://codesandbox.io/s/sad-field-c2p9p0.